libs/ml/include/ml/dataloaders/word2vec_loaders/sgns_w2v_dataloader.hpp

```diff
@@ -303,19 +374,23 @@ typename W2VLoader<T>::ReturnType W2VLoader<T>::GetNext()
 template <typename T>
 bool W2VLoader<T>::BuildVocab(std::string const &s)
 {
-  std::vector<SizeType> indexes = StringsToIndices(PreprocessString(s));
-  if (indexes.size() >=
-      2 * window_size_ + 1)  // each sentence stored in the data_ are guaranteed to have minimum
-                             // length to handle window_size context sampling
-  {
-    data_.push_back(std::move(indexes));
-    return true;
-  }
-  return false;
+    std::vector<std::string> preprocessed_string = PreprocessString(s);
+
+    if(preprocessed_string.size() <= 2*window_size_){ // dispose short sentences before we create vocabulary and count frequency: if we are not gonna train on it, the sentence does not exist
```

Comment:
this checks the size of one string? shouldn't you be checking the size of every sentence?